pagopa/io-app
# fix: [PE-732] CGN header snap point on scroll

## Short description
This pull request updates to the `CgnMerchantsListByCategory` component optimizing the layout for better user experience

## List of changes proposed in this pull request
- Adaptive padding calculation determined by the count of elements in a FlatList
- Replace deprecated component `GenericErrorComponent` with `OperationResultScreenContent`

## How to test
- Go to _Portafoglio_
- Tap on CGN
- Tap on _Scopri le opportunità_
- In `ts/features/bonus/cgn/screens/merchants/CgnMerchantsListByCategory.tsx` replace `merchantsAll` with this mock data function: 
```ts
const mockData = (count: number) =>
    Array.from({ length: count }, (_, i) => i.toString()).map(item => ({
      id: item,
      name: "Merchant " + item,
      imageUrl: "https://via.placeholder.com/150",
      discounts: [
        {
          id: "1",
          title: "Discount 1",
          description: "Description 1",
          discount: 10,
          discountType: "PERCENTAGE",
          discountValue: 10,
          startDate: "2021-06-01",
          endDate: "2021-06-30",
          termsAndConditions: "Terms and conditions 1",
          imageUrl: "https://via.placeholder.com/150"
        }
      ]
    }));
```
- After implementing the above changes, check the header behavior within the app to ensure it is functioning correctly. Make sure that it displays properly without any layout issues on scroll.

## Preview 

https://github.com/user-attachments/assets/0f6a38d5-3836-4273-a7d9-6a5e995e5c6a
